tools/codegen/api/types.py

```diff
@@ -218,6 +239,20 @@ def cpp_type_registration_declarations(self) -> str:
     def remove_const_ref(self) -> 'CType':
         return TupleCType([e.remove_const_ref() for e in self.elems])
 
+@dataclass(frozen=True)
+class VectorizedCType:
+    # Limited set of allowed specializations
```

Comment:
should we assert them at codegen time?

Reply:
I suppose we could, but it doesn't help all that much because typically VectorizedCType is passed a template argument, e.g.:

```
compute_t = VectorizedCType(BaseCType(scalar_t))
```

so we can't usefully do the check anyway (since the codegen doesn't know what the type actually is). I can improve the comment here to say how to find out which specializations are OK.